### ISSUE TITLE
Implement a scheduler class

### ISSUE THREAD

**Issue by MortenHolmRep** [graphnet-team/graphnet#415]
**Is your feature request related to a problem? Please describe.**
@RasmusOrsoe described a learning rate scheduler that may give better results. Previously @Aske-Rosted also mentioned other schedulers.
Implementing a class with scheduler options is a wanted feature and I have made a rough draft as an inspiration.

**Describe the solution you'd like**
I have drafted a potential code structure for a scheduler class. The code is untested and just made from looking at [this code](https://github.com/graphnet-team/graphnet/blob/main/examples/04_training/02_train_model_without_configs.py) on github; 
@asogaard @Aske-Rosted @RasmusOrsoe any thought on a potential structure? I have not considered the implementation with config files, but I do not think it will be an issue (?).

```
import torch.optim

class OptimizerScheduler:
    def __init__(self, optimizer_class, optimizer_kwargs, scheduler_class, scheduler_kwargs, scheduler_config):
        self.optimizer_class = optimizer_class
        self.optimizer_kwargs = optimizer_kwargs
        self.scheduler_class = scheduler_class
        self.scheduler_kwargs = scheduler_kwargs
        self.scheduler_config = scheduler_config
        
    def create(self, model):
        optimizer = self._create_optimizer(model)
        scheduler = self._create_scheduler(optimizer)
        return optimizer, scheduler
    
    def _create_optimizer(self, model):
        if self.optimizer_class == "Adam":
            optimizer = torch.optim.Adam(model.parameters(), **self.optimizer_kwargs)
        elif self.optimizer_class == "LBFGS":
            optimizer = torch.optim.LBFGS(model.parameters(), **self.optimizer_kwargs)
        else:
            raise NotImplementedError(f"Optimizer {self.optimizer_class} not implemented")
        return optimizer
    
    def _create_scheduler(self, optimizer):
        if self.scheduler_class == "PiecewiseLinearLR":
            scheduler = torch.optim.lr_scheduler.PiecewiseLinearLR(optimizer, **self.scheduler_kwargs)
        elif self.scheduler_class is None:
            scheduler = None
        else:
            raise NotImplementedError(f"Scheduler {self.scheduler_class} not implemented")
        return scheduler
```

The class would then be called as
```
optimizer_scheduler = OptimizerScheduler(
        optimizer_class=Adam,
        optimizer_kwargs={"lr": 1e-03, "eps": 1e-03},
        scheduler_class=PiecewiseLinearLR,
        scheduler_kwargs={
            "milestones": [
                0,
                len(training_dataloader) / 2,
                len(training_dataloader) * cast(int, config["n_epochs"]),
            ],
            "factors": [1e-2, 1, 1e-02],
        },
        scheduler_config={
            "interval": "step",
        },
    )

model = StandardModel(
        detector=detector,
        gnn=gnn,
        tasks=[task],
        optimizer_scheduler=optimizer_scheduler
    )
```

**Comment by asogaard** [comment]
PyTorch already seems to have a pretty rich set of learning rate schedulers, see [the list in this section](https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate). I think we should ask ourselves if the existing schedulers suit our needs, and if so whether we could make do with those. If not, we probably ought to stick as closely as possible to the structure of existing schedulers, see e.g. `LRScheduler` [here](https://pytorch.org/docs/stable/_modules/torch/optim/lr_scheduler.html#LRScheduler).
